### ISSUE TITLE
memleaks in X509_REQ_to_X509, and X509_REQ_to_X509 not completely.

### ISSUE THREAD

**Issue by zhaozg** [openssl/openssl#182]
When I loop with X509_REQ_to_X509, memory grows infinite, 
And I see extension assign not finished.

Any plan on it.

```c
X509 *X509_REQ_to_X509(X509_REQ *r, int days, EVP_PKEY *pkey)
	{
	X509 *ret=NULL;
	X509_CINF *xi=NULL;
	X509_NAME *xn;

	if ((ret=X509_new()) == NULL)
		{
		X509err(X509_F_X509_REQ_TO_X509,ERR_R_MALLOC_FAILURE);
		goto err;
		}

	/* duplicate the request */
	xi=ret->cert_info;

	if (sk_X509_ATTRIBUTE_num(r->req_info->attributes) != 0)
		{
		if ((xi->version=M_ASN1_INTEGER_new()) == NULL) goto err;
		if (!ASN1_INTEGER_set(xi->version,2)) goto err;
/*		xi->extensions=ri->attributes; <- bad, should not ever be done
		ri->attributes=NULL; */
		}

	xn=X509_REQ_get_subject_name(r);
	if (X509_set_subject_name(ret,X509_NAME_dup(xn)) == 0)
		goto err;
	if (X509_set_issuer_name(ret,X509_NAME_dup(xn)) == 0)
		goto err;

	if (X509_gmtime_adj(xi->validity->notBefore,0) == NULL)
		goto err;
	if (X509_gmtime_adj(xi->validity->notAfter,(long)60*60*24*days) == NULL)
		goto err;

	X509_set_pubkey(ret,X509_REQ_get_pubkey(r));

	if (!X509_sign(ret,pkey,EVP_md5()))
		goto err;
	if (0)
		{
err:
		X509_free(ret);
		ret=NULL;
		}
	return(ret);
	}
```

**Comment by nick599** [comment]
@zhaozg do you still have a problem with this? - No comments on this since Oct 2014

**Comment by nick599** [comment]
@richsalz what should be the next steps with this issue? does the proposed code need to be reviewed? What should we do when something like this happens?

**Comment by richsalz** [comment]
It's a bug/PR that nobody has looked at recently.  We got lots of those, sadly.

**Comment by FdaSilvaYY** [comment]
Hi,
I dig this one a bit.

X509_REQ_to_X50O is leaking by calling X509_NAME_dup, 
because X509_set_subject/issuer_name is also doing the same.

Moreover, IMHO, req_main is not using memleaks test harness, that why it is unseen.

I will try to fix it on my repo, before submitting a PR.

**Comment by mattcaswell** [comment]
This got fixed by #933. Closing.

**Comment by FdaSilvaYY** [comment]
Backport to 1.0.2 branch is waiting for commit , see PR #999
